base/c.jl

```diff
@@ -23,7 +23,7 @@ if is_windows()
 else
     typealias Clong Int
     typealias Culong UInt
-    typealias Cwchar_t Int32
+    typealias Cwchar_t UInt32
```

Comment:
I'm not expertise on it, should I reverse it to `Int32`? Aren't the used to represent characters and therefore negatives are not useful?

Maybe `Int32` is faster than `UInt32` even if `UInt32` doesn't break anything.